src/Inventory/Conf.php

```diff
@@ -142,48 +142,34 @@ public function knownInventoryExtensions(): array
     }
 
     /**
-     * Import inventory file
+     * Import inventory files
      *
-     * @param array $files $_FILES
+     * @param array $files[filename => filepath] Files to import
      *
-     * @return Request
+     * @return array [filename => [success => bool, message => string, asset => CommonDBTM]]
      */
-    public function importFile($files): Request
+    public function importFiles($files): array
     {
-        $path = $files['inventory_file']['tmp_name'];
-        $name = $files['inventory_file']['name'];
-
-        $inventory_request = new Request();
-
-        if ($this->isInventoryFile($name)) {
-           //knwon standalone file type, try to import.
-            $contents = file_get_contents($path);
-            $this->importContentFile($inventory_request, $path, $contents);
-            return $inventory_request;
-        }
-
-       //was not a known file, maybe an archive
-        $archive = UnifiedArchive::open($path);
-        if ($archive === null) {
-           //nay, not an archive neither
-            Session::addMessageAfterRedirect(
-                __('No file to import!'),
-                false,
-                ERROR
-            );
-            return $inventory_request;
-        }
+        $result = [];
 
-       //process archive
-        $files = $archive->getFileNames();
-        foreach ($files as $file) {
-            if ($this->isInventoryFile($file)) {
-                $contents = $archive->getFileContent($file);
-                $this->importContentFile($inventory_request, null, $contents);
+        foreach ($files as $filename => $filepath) {
+            $archive = UnifiedArchive::open($filepath);
+            if ($archive === null) {
+                if ($this->isInventoryFile($filename)) {
+                    $result[$filename] = $this->importContentFile($filepath, file_get_contents($filepath));
+                }
+            } else {
+                $unarchived_files = $archive->getFileNames();
```

Comment:
looks like in 10.1 PR you used `getFiles`; maybe that explains tests failures
```suggestion
                $unarchived_files = $archive->getFile();
```